R/tm_a_regression.R

```diff
@@ -241,20 +245,57 @@ srv_a_regression <- function(id,
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
-    anl_merged_input <- teal.transform::merge_expression_module(
+
+    rule_rvr1 <- function(value) {
+      if (isTRUE(input$plot_type == "Response vs Regressor")) {
+        if (length(value) > 1L)
+          "This plot can only have one regressor."
+      }
+    }
+    rule_rvr2 <- function(other) {
+      function(value) {
+        if (isTRUE(input$plot_type == "Response vs Regressor")) {
+          otherval <- selector_list()[[other]]()$select
+          if (isTRUE(value == otherval))
+            "Response and Regressor must be different."
+        }
+      }
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = list(response = response, regressor = regressor),
       datasets = data,
-      join_keys = get_join_keys(data),
-      data_extract = list(response = response, regressor = regressor)
+      select_validation_rule = list(
+        regressor = shinyvalidate::compose_rules(
+          shinyvalidate::sv_required("At least one regressor should be selected."),
+          rule_rvr1,
+          rule_rvr2("response")
+        ),
+        response = shinyvalidate::compose_rules(
+          shinyvalidate::sv_required("At least one response should be selected."),
+          rule_rvr2("regressor")
+        )
+      )
+    )
+
+    iv_r <- reactive({
+      iv <- shinyvalidate::InputValidator$new()
+      teal.transform::compose_and_enable_validators(iv, selector_list)
+    })
+
+    iv_out <- shinyvalidate::InputValidator$new()
+    iv_out$condition(~ isTRUE(input$show_outlier))
+    iv_out$add_rule("label_var", shinyvalidate::sv_required("Please provide an `Outlier label` variable"))
+    iv_out$enable()
```

Comment:
```suggestion
    iv_r <- reactive({
      iv <- shinyvalidate::InputValidator$new()
      iv_out <- shinyvalidate::InputValidator$new()
      iv_out$condition(~ isTRUE(input$show_outlier))
      iv_out$add_rule("label_var", shinyvalidate::sv_required("Please provide an `Outlier label` variable"))
      iv$add_validator(iv_out)
      teal.transform::compose_and_enable_validators(iv, selector_list)
    })
```

Reply:
I would keep them separate if the outlier label wasn't required to generate the output but it looks like it is?
![image](https://user-images.githubusercontent.com/15201933/210364181-88c8be19-cdbb-41d1-bd5a-4888704416ac.png)

Reply:
Did you test this? It's the only way I could make the reactivity work.

Reply:
no I didn't try it...

Reply:
Upon closer inspection, we decided to leave it as is.